src/png.imageio/pnginput.cpp

```diff
@@ -162,7 +162,8 @@ PNGInput::open(const std::string& name, ImageSpec& newspec)
     bool ok = PNG_pvt::read_info(m_png, m_info, m_bit_depth, m_color_type,
                                  m_interlace_type, m_bg, m_spec,
                                  m_keep_unassociated_alpha);
-    if (!ok || m_err) {
+    if (!ok || m_err
+        || check_open(m_spec, { 0, 1 << 30, 0, 1 << 30, 0, 1, 0, 4 })) {
```

Comment:
I am not sure what is a reasonable max size here is. According to the doc, the width and height are four-byte unsigned integer. So the range is range 0 to 2^31 - 1.

Reply:
For OpenEXR output, here is what we did, maybe that's the right approach here too?

        // Arbitrarily limit res to 1M x 1M and 4k channels, assuming anything
        // beyond that is more likely to be a mistake than a legit request. We
        // may have to come back to this if these assumptions are wrong.
        if (!check_open(Create, srcspec,
                        { 0, 1 << 20, 0, 1 << 20, 0, 1, 0, 1 << 12 }))
            return false;

So basically, we just decided that although it *could* be a full uint32 size, we're going to assume nobody needs ore than 1M pixels in each direction.

If that makes us nervous, then as a next step, perhaps instead of hard-coding it, we could make a global OIIO attribute that provides a reasonable resolution limit, defaulting to 1M say, and then if somebody really has a legit need for larger dimensions, there is a way to override it without changing the source.

Reply:
Okay than I think I will try to match the outputs limits (if they exist). And yea, 1 << 30 might be bit too big for general use cases.